src/component/addons/AddonList/ConfiguredAddons/ConfiguredAddonsActionCell/ConfiguredAddonsActionsCell.tsx

```diff
@@ -23,19 +25,24 @@ export const ConfiguredAddonsActionsCell = ({
     original,
 }: IConfiguredAddonsActionsCellProps) => {
     const navigate = useNavigate();
+    const [isEnabled, setIsEnabled, rollbackIsChecked] =
+        useOptimisticUpdate<boolean>(original.enabled);
+
+    const onClick = () => {
+        setIsEnabled(!isEnabled);
+        toggleAddon(original).catch(rollbackIsChecked);
+    };
+
     return (
         <ActionCell>
-            <PermissionIconButton
-                permission={UPDATE_ADDON}
-                onClick={() => toggleAddon(original)}
-                tooltipProps={{ title: 'Toggle addon' }}
-            >
-                <ConditionallyRender
-                    condition={original.enabled}
-                    show={<Visibility />}
-                    elseShow={<VisibilityOff />}
+            <Tooltip title={isEnabled ? 'Disable' : 'Enable'} arrow>
+                <PermissionSwitch
```

Comment:
Doesn't look like the `input` gets a label, should fix that somehow. Maybe MUI has a way.